### PR TITLE
Item interface + fix bottom bar display

### PR DESCRIPTION
**Pull request by 3nids**
This makes the code a bit clearer (at least to me) by explicitly having interfaces rather than floating lambdas.
This also removes a sip code (interesting though) that is able to run python lambdas.

this also fixes the bottom bar not being scaled up. I switched it to QgsFloatingWidget at the same time.

### CODE REVIEW

File: kadas/app/milx/kadasmilxintegration.cpp [hunk 1]
```diff
@@ -159,15 +166,10 @@ void KadasMilxIntegration::createMilx( bool active )
...
-
-    KadasMapToolCreateItem::ItemFactory itemFactory = [ = ]
-    {
-      return new KadasMilxItem();
-    };
     KadasLayerSelectionWidget::LayerFilter layerFilter = [ = ]( QgsMapLayer * layer ) { return dynamic_cast<KadasMilxLayer *>( layer ); };
     KadasLayerSelectionWidget::LayerCreator layerCreator = [ = ]( const QString & name ) { return new KadasMilxLayer( name ); };
 
-    KadasMapToolCreateItem *tool = new KadasMapToolCreateItem( canvas, itemFactory, getOrCreateLayer() );
+    KadasMapToolCreateItem *tool = new KadasMapToolCreateItem( canvas, new KadasMilxInterface(), getOrCreateLayer() );
```
Comment (m-kuhn) [inline kadas/app/milx/kadasmilxintegration.cpp:172]: Is the interface ever deleted?
Possibly make ownership clear by using a `std::unique_ptr` in the item and `std::move` it in?

**Comment by m-kuhn** [comment]
Some good cleanup in here !
I think there's a memory leak around `KadasMilxInterface`, once that's cleaned, it's good to go IMHO.

File: kadas/app/milx/kadasmilxintegration.cpp (hunk 1 above)
Comment (3nids) [reply kadas/app/milx/kadasmilxintegration.cpp:172]: it is deleted in KadasMapToolCreateItem::~KadasMapToolCreateItem()

is your approach safer / cleaner?

Comment (m-kuhn) [reply kadas/app/milx/kadasmilxintegration.cpp:172]: safer and cleaner yes (initialization already guarantees that it's deleted) but if it's deleted in the destructor, that's also ok

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**